packages/files-ui/src/Components/Modules/FileBrowsers/UploadFileModal.tsx

```diff
@@ -83,8 +83,19 @@ const UploadFileModule = ({ modalOpen, close }: IUploadFileModuleProps) => {
   const [isDoneDisabled, setIsDoneDisabled] = useState(true)
   const { uploadFiles } = useFiles()
   const { currentPath, refreshContents, bucket } = useFileBrowser()
+  const { storageSummary } = useFiles()
 
-  const UploadSchema = object().shape({ files: array().required(t`Please select a file to upload`) })
+  const UploadSchema = useMemo(() => object().shape({
+    files: array().required(t`Please select a file to upload`)
+      .test("Upload Size",
+        t`Upload size exceeds plan capacity`,
+        (files) => {
+          const isOwner = bucket?.permission === "owner"
+          const availableStorage = storageSummary?.available_storage || 0
+          const uploadSize = files?.reduce((total: number, file: File) => total += file.size, 0) || 0
+          return !(isOwner && uploadSize > availableStorage)
```

Comment:
I find it easier to read with an early return
```suggestion
          // no validation if the user isn't the owner
          if(bucket?.permission !== "owner") return true
          
          const availableStorage = storageSummary?.available_storage || 0
          const uploadSize = files?.reduce((total: number, file: File) => total += file.size, 0) || 0
          return uploadSize < availableStorage
```